src/engine/tile/tileset.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <external/SDL2/include/SDL.h>
+#include <stdio.h>
+
+struct Tileset 
+{
+    Uint32 tile_size;
+    Uint32 n_rows;
+    Uint32 n_cols;
+    Uint8 *m_Bitmask;
```

Comment:
probably use a std::vector here. It should also take care of cleaning up the memory once it goes out of scope, so no need to worry about `free()`ing `malloc()`ed data

Reply:
std::vector has a function to set the capacity on initialization